File: src/main/kotlin/com/chopeks/psi/impl/SquirrelDocCommentImpl.java
package com.chopeks.psi.impl;

import com.chopeks.psi.SquirrelDocComment;
import com.intellij.extapi.psi.ASTWrapperPsiElement;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElement;
import com.intellij.psi.tree.IElementType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class SquirrelDocCommentImpl extends ASTWrapperPsiElement implements SquirrelDocComment {

    public SquirrelDocCommentImpl(@NotNull final ASTNode node) {
        super(node);
    }

    @Nullable
    public PsiElement getOwner() {
        return null;
    }

    public IElementType getTokenType() {
        return getNode().getElementType();
    }
}

File: src/main/kotlin/com/chopeks/psi/impl/SquirrelDocCommentImpl.kt
package com.chopeks.psi.impl

import com.chopeks.psi.SquirrelDocComment
import com.intellij.extapi.psi.ASTWrapperPsiElement
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement

class SquirrelDocCommentImpl(node: ASTNode) : ASTWrapperPsiElement(node), SquirrelDocComment {
	override fun getOwner(): PsiElement? = null
	override fun getTokenType() = node.elementType
}